client/src/components/NotFound/tests/index.test.tsx

```diff
@@ -1,16 +1,42 @@
-import { mount, shallow } from 'enzyme';
-import toJson from 'enzyme-to-json';
+import { mount } from 'enzyme';
 import React from 'react';
+import { MemoryRouter, Route, Switch } from 'react-router-dom';
+import NotFound from '..';
 
-import NotFound from '../';
+const NotFoundLocation = 'pageurlwhichisnotavalible';
 
-describe('components/NotFount', () => {
-  it('renders without crashing', () => {
-    shallow(<NotFound />);
+describe('src/components/NotFound', () => {
+  it('renders correctly', () => {
+    const wrapper = mount(
+      <MemoryRouter
+        initialEntries={[{ pathname: NotFoundLocation, search: '', hash: '', state: {} }]}
+      >
+        <NotFound />
+      </MemoryRouter>
+    );
+    // should redirect to home
+    expect(wrapper.find('.ant-result-404')).toHaveLength(1);
   });
-  it('renders the NotFount component', () => {
-    const wrapper = mount(<NotFound />);
-    expect(toJson(wrapper)).toMatchSnapshot();
-    wrapper.unmount();
+
+  it('redirects to home on button click', () => {
+    const wrapper = mount(
+      <MemoryRouter
+        initialEntries={[{ pathname: NotFoundLocation, search: '', hash: '', state: {} }]}
+      >
+        <Switch>
+          {/* tslint:disable-next-line: jsx-no-lambda */}
+          <Route path="/" component={() => <div id="home" />} />
+          {/* tslint:disable-next-line: jsx-no-lambda */}
+          <Route component={() => <NotFound />} />
+        </Switch>
+      </MemoryRouter>
+    );
+
+    // should redirect to teams
+    expect(wrapper.find('.ant-result-404')).toHaveLength(1);
+    wrapper.find('button').simulate('click');
+
+    expect(wrapper.find('.ant-result-404')).toHaveLength(0);
+    expect(wrapper.find('#home')).toHaveLength(1);
```

Comment:
Nice tests @ashfahan. Though in courtesy of @moshthepitt on article https://kentcdodds.com/blog/effective-snapshot-testing, We can make our work easier, by testing the props passed to component `Result` since we should make the assumption that component `Result` is properly tested and we should not be worried about  its contents